tests/transform/test_genes_biodomains.py

```diff
@@ -40,57 +46,51 @@ def test_count_grouped_total_two_groups(self):
-#             {
-#                 "ensembl_gene_id": ["1", "1", "2", "2", "3", "3"],
-#                 "biodomain": ["a", "b", "c", "d", "e", "f"],
-#                 "go_terms": ["a", "b", "c", "d", "e", "f"],
-#             }
-#         )
-#     }
-#     expected_gene_biodomains_col = [
-#         [{"biodomain": "a", "go_terms": ["a"]}, {"biodomain": "b", "go_terms": ["b"]}],
-#         [{"biodomain": "c", "go_terms": ["c"]}, {"biodomain": "d", "go_terms": ["d"]}],
-#         [{"biodomain": "e", "go_terms": ["e"]}, {"biodomain": "f", "go_terms": ["f"]}],
-#     ]
-#     test_biodomains = transform.transform_biodomains(datasets=test_datasets)
-#     assert list(test_biodomains["gene_biodomains"]) == expected_gene_biodomains_col
-
-
-# df = pd.DataFrame(
-#     {'team id': [np.nan, 0, 1, 2],
-#      'team.Name': ['MSN', 'Team 1', 'Team 2', np.nan],
-#      'team-Sco@#&': ['x', 'y', 'z', "na"]})
-
-# def test_standardize_column_names():
-
-#     result_df = transform.standardize_column_names(df)
-#     assert type(result_df) is pd.core.frame.DataFrame
-#     assert list(result_df.columns) == ['team_id', 'team_name', 'team-sco']
-
-
-# def test_standardize_values():
-
-#     assert df.isna().sum().sum() == 2
-
-#     result_df = transform.standardize_values(df)
-
-#     assert type(result_df) is pd.core.frame.DataFrame
-#     assert result_df.isna().sum().sum() == 0
-#     assert result_df.shape == (4, 3)
-
-# def test_rename_columns():
-#     refresh_df = pd.DataFrame(
-#         {'team id': [np.nan, 0, 1, 2],
-#          'team.Name': ['MSN', 'Team 1', 'Team 2', np.nan],
-#          'team-Sco@#&': ['x', 'y', 'z', "na"]})
-
-#     bad_result_df = transform.rename_columns(df=refresh_df, column_map={"team-Sco@#&"})
-#     assert type(bad_result_df) is pd.core.frame.DataFrame
-#     assert list(bad_result_df.columns) == ["team id", "team.Name", "team-Sco@#&"]
-
-#     partial_good_result_df = transform.rename_columns(df=refresh_df, column_map={"team-Sco@#&": "team_scope"})
-#     assert list(partial_good_result_df.columns) == ['team id', 'team.Name', 'team_scope']
-#     assert type(partial_good_result_df) is pd.core.frame.DataFrame
+class TestTransformBiodomains:
+    data_files_path = "tests/test_assets/data_files"
+    test_data = [
+        (
+            "biodomains_test_input.csv",
+            "genes_biodomains.json",
+            True,
+        ),  # pass with good data
+        (  # pass with imperfect, but handled data
+            "biodomains_test_input_bad_but_should_pass.csv",
+            "genes_biodomains_bad_output_but_should_pass.json",
+            True,
+        ),
+        (
+            "biodomains_test_input_bad_should_fail.csv",
+            None,
+            False,
+        ),  # fail with bad data
+    ]
+
+    @pytest.mark.parametrize("input_file, expected_output_file, expect_pass", test_data)
+    def test_transform_biodomains(self, input_file, expected_output_file, expect_pass):
+        if expect_pass:
```

Comment:
I like what you're thinking here with merging the test in one test function by expect_pass, but my recommendation is to split the requirements in different functions

1. Test that function passes
2. Test that function fails